File: Vue-App_Basics/src/utils/dish.ts
export interface Dish {
	Id?: string;
	Name?: string;
	Description?: string;
	Price?: number;
	category?: DishCategory;
	IsAvailable?: boolean;
	availability?: Availability[];
	TimeToCook?: number;
}

export enum Availability{
	Breakfast,
	Lunch,
	Dinner,
	WeekDays,
	Weekend,
}

export enum DishCategory
{
	Starter,
	Main_course,
	Dessert,
	Beverage,
	Other,

}

File: Models/Dish.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace DigitalMenuRestaurant.Models
{
    public class Dish
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        public string Name { get; set; }
        public string Description { get; set; }
        public double Price { get; set; }
        public DishCategory category { get; set; }
        public List<Availability> availability { get; set; }
        public bool IsAvailable { get; set; }
        //Time is stored in minutes
        public int TimeToCook { get; set; }
    }
}
